### DISCUSSION TITLE
i3 no longer considering dvorak keyboard layout

### DISCUSSION THREAD

**Discussion by rpdelaney** [i3/i3#4693] [Q&A] [answered]
```
$ i3 --version
i3 version 4.20.1 (2021-11-03)
```
I use `setxkbmap dvorak`. My binds assume this layout. Recently, the binds are behaving as if my layout is `us` (qwerty), although I am still typing in dvorak everywhere else. This is disorienting. Is this intended behavior? I do not want to rebind everything if this is a bug that will be fixed later. Was some setting added that I need to set in the config? I was not able to find anything about this after scanning the changelog.

**Comment by Airblader** [comment, marked as answer]
No, there are no changes that should cause this. Please file a bug report. Does downgrading to 4.19 fix it?

**Reply by rpdelaney** [reply]
OK, I will try downgrading when I am back at my computer and I will file a bug report if that fixes it.

**Reply by rpdelaney** [reply]
It seems to have resolved itself after a restart. Not sure what the issue was.

**Reply by lilibyte** [reply]
not an i3 issue, but this happens to me when i unplug my keyboard and plug it back in. should be fixed by adding persistent configuration like [X config files](https://wiki.archlinux.org/title/Xorg/Keyboard_configuration#Using_X_configuration_files) or in [`/etc/vconsole.conf`](https://wiki.archlinux.org/title/Linux_console/Keyboard_configuration#Persistent_configuration) for tty (may need a [`keymap` hook in your mkinitcpio config](https://wiki.archlinux.org/title/Mkinitcpio#Common_hooks))

if it happens again share the output of `localectl status`
